src/read.c

```diff
@@ -2701,11 +2674,7 @@ UInt ReadEvalFile(Obj *evalResult)
     if (TLS(CurrentHashLock))
       HashUnlock(TLS(CurrentHashLock));
 #endif
-    rs->StackNams    = stackNams;
-    rs->ReadTop      = readTop;
-    rs->ReadTilde    = readTilde;
     STATE(Tilde)     = tilde;
-    rs->CurrLHSGVar  = currLHSGVar;
```

Comment:
Real change: since we freshly allocated the `ReaderState` on the stack, there is no need to "restore" its content before returning.